### PR TITLE
Adding "HandshakeHandler" and "DispatchToCustomSequence" functionality

### PR DESCRIPTION
**Pull request by Arshardh**
Add HandshakeHandler and DispatchToCustomSequence functionality to websocket inbound endpoints.

### CODE REVIEW

**Comment by djkevincr** [comment]
@Arshardh Wouldn't it be better to implement channel pipeline handler generic here? Rather than saying HandshakeHandler, we may use PipelineHandler. I think this will be good extension point for adding Netty custom pipeline handler if one use requires, not only for just handle the alteration of default handshake handling mechanism. 
To me code is written generic, there are no case where it handle specific logic that restricted to handle custom handshake? Only the variable names and classes you name implies the usage of altering handshake.
It is good if you can add example handshake handler class to examples here similar to EchoSubprotocolHandler.
Anyway I am +1 to merge changes

**Comment by ramith** [comment]
Please consider squashing the PR before pushing?

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/PipelineHandlerBuilderUtil.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+
+import java.lang.reflect.Constructor;
+
+public class PipelineHandlerBuilderUtil {
+    private static final Log log = LogFactory.getLog(PipelineHandlerBuilderUtil.class);
+
+    public static ChannelHandler stringToPipelineHandlers(String handlerClass) {
```
Comment (ramith) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/PipelineHandlerBuilderUtil.java]: what does this method do? could you explain ? (that would add value to code you are contributing).

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/pipelinehandlers/SamplePipelineHandler.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.inbound.endpoint.protocol.websocket.pipelinehandlers;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+public class SamplePipelineHandler extends ChannelInboundHandlerAdapter {
```
Comment (ramith) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/pipelinehandlers/SamplePipelineHandler.java]: explain what this class is in a java doc? (team members who maintain the code will find it definitely useful)

**Pull request merged**